src/ring_theory/localization/basic.lean

```diff
@@ -989,16 +989,25 @@ variables (S M) (Q : Type*) [comm_ring Q] {g : R →+* P} [algebra P Q]
 
 /-- Injectivity of a map descends to the map induced on localizations. -/
 lemma map_injective_of_injective
-  (hg : function.injective g) [is_localization (M.map g : submonoid P) Q]
-  (hM : (M.map g : submonoid P) ≤ non_zero_divisors P) :
+  (hg : function.injective g) [is_localization (M.map g : submonoid P) Q] :
   function.injective (map Q g M.le_comap_map : S → Q) :=
 begin
   rintros x y hxy,
-  obtain ⟨a, b, rfl⟩ := mk'_surjective M x,
-  obtain ⟨c, d, rfl⟩ := mk'_surjective M y,
-  rw [map_mk' _ a b, map_mk' _ c d, mk'_eq_iff_eq] at hxy,
-  refine mk'_eq_iff_eq.2 (congr_arg (algebra_map _ _) (hg _)),
-  convert is_localization.injective _ hM hxy; simp,
+  rw ← sub_eq_zero at ⊢ hxy,
+  rw ← map_sub at hxy,
+  set z := x - y,
+  obtain ⟨a, b, hz⟩ := is_localization.mk'_surjective M z,
+  rw [← hz, is_localization.map_mk', is_localization.mk'_eq_zero_iff] at hxy,
```

Comment:
This is a little shorter:
```suggestion
  rw injective_iff_map_eq_zero,
  intros z hxy,
  obtain ⟨a, b, rfl⟩ := is_localization.mk'_surjective M z,
  rw [is_localization.map_mk', is_localization.mk'_eq_zero_iff] at hxy,
```

Reply:
Thanks, great suggestion! I applied it and changed the `hxy` to `hz` everywhere/